File: src/test/java/org/hisp/dhis/rules/models/RuleVariableNewestEventTest.java
package org.hisp.dhis.rules.models;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import java.util.ArrayList;

import static org.junit.Assert.assertEquals;

@RunWith( JUnit4.class )
public class RuleVariableNewestEventTest
{
    @Test
    public void createShouldPropagatePropertiesCorrectly()
    {
        RuleVariableNewestEvent ruleVariableNewestEvent = RuleVariableNewestEvent.create(
            "test_variable", "test_dataelement", RuleValueType.NUMERIC, true, new ArrayList<>());

        assertEquals( "test_variable" , ruleVariableNewestEvent.name() );
        assertEquals( "test_dataelement" , ruleVariableNewestEvent.dataElement() );
        assertEquals( RuleValueType.NUMERIC , ruleVariableNewestEvent.dataElementType() );
    }
}

File: src/jvmTest/kotlin/org/hisp/dhis/rules/models/RuleVariableNewestEventTest.kt
package org.hisp.dhis.rules.models

import kotlin.test.Test
import kotlin.test.assertEquals
class RuleVariableNewestEventTest {
    @Test
    fun createShouldPropagatePropertiesCorrectly() {
        val ruleVariableNewestEvent = RuleVariableNewestEvent(
            "test_variable", true, ArrayList(), "test_dataelement", RuleValueType.NUMERIC
        )
        assertEquals("test_variable", ruleVariableNewestEvent.name)
        assertEquals("test_dataelement", ruleVariableNewestEvent.dataElement)
        assertEquals(org.hisp.dhis.rules.models.RuleValueType.NUMERIC, ruleVariableNewestEvent.dataElementType)
    }
}
